addons/mail/static/src/models/message/message.js

```diff
@@ -378,6 +378,16 @@ function factory(dependencies) {
             return timeFromNow(this.date);
         }
 
+        /**
+         * @returns {boolean}
+         */
+        _computeCleanedSubject() {
+            if (this.subject && this.originThread) {
+                const subject = `Re: ${this.originThread.name}`;
+                return !(this.subject === subject || this.subject.endsWith(subject));
+            }
+        }
```

Comment:
This is not exactly what I was thinking.
What I had in mind is to have this compute make the cleaned subject string, so remove the useless Re: from it.

And then in the view, we can compare thread name != cleaned subject